code/drasil-theory/lib/Theory/Drasil/ConstraintSet.hs

```diff
@@ -14,13 +14,14 @@ import qualified Data.List.NonEmpty as NE
 
 -- | 'ConstraintSet's are sets of invariants that always hold for underlying domains.
 data ConstraintSet e = CL {
-    _con  :: ConceptChunk,
-    _invs :: NE.NonEmpty e
+    _csUid :: UID,
```

Comment:
So this doesn't seem quite right [but I'm not sure]. It feels like the `ConceptChunk` should be the one that is created to exactly mean this set of constraints. So there shouldn't be 2 UIDs, because there shouldn't be 2 different ideas.

It could also be that this is truly a different idea, where we're associating a set of constraints to an existing idea, and so this association does form something new.

Reply:
I think this heavily relies on how we want to build chunks. In #2873, I viewed Chunks as self-contained units of knowledge and be denoted by a single type with properties shown through typeclasses. The units of knowledge might be built on other chunks (e.g., as it is here, forming a "constraint" relationship between a concept chunk "target" and a list of `e` "constraints". Another example might be QDefinitions where a QDefinition is a specific relationship between a QuantityDict and a singular defining expression), or pushed out from others (e.g., commonly IdeaDicts).

I guess we should pause work on #2873 because that assumes a single Chunk is a single self-contained unit of knowledge. If it isn't (and we want to have a dynamic map-like structure for chunks), then we _might_ need to change what the new ChunkDBs will look like.

Reply:
I agree, we need to get the idea and design of Chunks (and UIDs) straight first.